File: 1899-count-items-matching-a-rule/1899-count-items-matching-a-rule.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        int row = items.size() ;
        int cnt = 0 ;
        int ind = 0 ;
        if(ruleKey == "type")   ind = 0 ;
        else if(ruleKey == "color") ind = 1 ;
        else ind = 2 ;
        for(int i = 0 ; i < row ; i++){
            if(items[i][ind] == ruleValue)  cnt++ ;
        }
        return cnt ;
    }
};

File: 1899-count-items-matching-a-rule/1899-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int row = items.size() ;
        int cnt = 0 ;
        int ind = 0 ;
        switch(ruleKey){
            case "type" : ind = 0 ;    break ;
            case "color" : ind = 1 ;    break ;
            case "name" : ind = 2 ;    break ;
        }
        for(List item : items){
            if(item.get(ind).equals(ruleValue)) cnt++ ;
        }
        return cnt ;
    }
}